tests/hipscat/io/test_write_metadata.py

```diff
@@ -40,6 +41,25 @@ def test_write_json_file(assert_text_file_matches, tmp_path):
     assert_text_file_matches(expected_lines, json_filename)
 
 
+def test_write_json_paths(assert_text_file_matches, tmp_path):
+    posix_path = Path(tmp_path)
```

Comment:
FYI, `Path` is `PosixPath` on Unix and `WindowsPath` on Windows
https://docs.python.org/3/library/pathlib.html